sched/semaphore/spinlock.c

```diff
@@ -474,7 +474,7 @@ void spin_clrbit(FAR volatile cpu_set_t *set, unsigned int cpu,
 
 void read_lock(FAR volatile rwlock_t *lock)
 {
-  rwlock_t old;
+  int32_t old;
```

Comment:
why not continue rwlock_t

Reply:
Just for pass Mac sim-02 ci

Reply:
ok, should we use int instead int32_t and please merge with line 481